gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageTest.java

```diff
@@ -3379,6 +3383,94 @@ public void initializeRequest_withDownscopedAccessToken() throws IOException {
         .isEqualTo("Bearer testDownscopedAccessToken");
   }
 
+  @Test
+  public void checkDeletionOrderForHnBucketBalancedFolders() throws InterruptedException {
+    String folderString = "test-folder-start/";
+    List<FolderInfo> foldersToDelete = new LinkedList<>();
+    List<FolderInfo> orderOfDeletion = new LinkedList<>();
+
+    addFolders(foldersToDelete, folderString);
+    int folderSize = foldersToDelete.size();
+    DeleteFolderOperation deleteFolderOperation =
+        new DeleteFolderOperation(foldersToDelete, GoogleCloudStorageOptions.DEFAULT, null);
+
+    deleteFolderOperation.computeChildrenForFolderResource();
+    while (folderSize != 0 && deleteFolderOperation.encounteredNoExceptions()) {
+      FolderInfo folderToDelete = deleteFolderOperation.getElementFromBlockingQueue();
+      orderOfDeletion.add(folderToDelete);
+      folderSize--;
+      deleteFolderOperation.addToToBatchExecutorQueue(() -> null, folderToDelete, 1);
```

Comment:
Why is this having null as callback?

Reply:
We are asserting that the callback cannot be null, correct? Why not add a test callback and validate that too?